src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -4938,6 +4941,30 @@ private static BigDecimal createAndStripZerosToMatchScale(BigInteger intVal, int
         return valueOf(intVal, scale, 0);
     }
 
+    private static BigDecimal createAndStripZerosToMatchScaleFast(BigInteger intVal, int scale, long preferredScale) {
+        BigInteger qr[]; // quotient-remainder pair
+        int scaleStep;
+        while (intVal.compareMagnitude(BigInteger.TEN) >= 0
+                && scale > preferredScale) {
+            scaleStep = checkScale(intVal, Math.max(((long) scale - preferredScale) / 2, 1));
+            if (intVal.getLowestSetBit() >= scaleStep) {
```

Comment:
I think there's no need to recompute `intVal.getLowestSetBit()` at each iteration.
If you store the original one in an `int` variable `lsb` initialized upon entry in the method, then you can simply compute `lsb -= scaleStep` at each iteration. It can become negative, but that shouldn't be a problem.

What do you think?